### PR TITLE
Name variables in callbacks

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

**Comment by ericmj** [comment]
Thank you.

File: lib/elixir/lib/calendar.ex [hunk 1]
```diff
@@ -67,7 +67,7 @@ defmodule Calendar do
   @doc """
   Converts the given structure into a string according to the calendar.
   """
-  @callback to_string(Date.t | NaiveDateTime.t | DateTime.t) :: String.t
+  @callback to_string(structure) :: String.t when structure: Date.t | DateTime.t | NaiveDateTime.t
```
Comment (lexmag) [inline lib/elixir/lib/calendar.ex]: Personally I find previous definition more readable.

Comment (michalmuskala) [inline lib/elixir/lib/calendar.ex]: I think ex_doc gives names like `arg0` to the callback variables, unless they are named in the `@callback` specification.

We had this in Ecto.

Comment (whatyouhide) [inline lib/elixir/lib/calendar.ex]: Maybe 

```elixir
@callback to_string(structure :: Date.t | ...)
```

I am 👎 on `when` as well, looks unnecessarily complex here.

Comment (lexmag) [inline lib/elixir/lib/calendar.ex]: @michalmuskala you're right, it will be `arg0` without this change.

Comment (eksperimental) [inline lib/elixir/lib/calendar.ex]: Sorry i wasnt clear. The whole point of the pr was to avoid arg0. I can submit a new pr with whatyouhide version

**Pull request merged**